src/190B20.c

```diff
@@ -1135,7 +1135,26 @@ INCLUDE_ASM(s32, "190B20", func_802636E4);
 
 INCLUDE_ASM(s32, "190B20", func_80263914);
 
-INCLUDE_ASM(s32, "190B20", count_power_plus);
+s32 count_power_plus(s32 arg0) {
+    s32 i;
+    u8 moveID;
+    s32 pp;
+
+    if (gGameStatusPtr->peachFlags & 1) {
+        return 0;
+    }
+
+    for (pp = 0, i = 0; i < ARRAY_COUNT(gPlayerData.equippedBadges); i++) {
```

Comment:
I'd prefer pp be set to 0 outside of the loop setup, because it's not really a loop variable in my eyes.. It only gets modified if certain conditions are met, and its scope is beyond the loop. Could we move it to the line before the loop start?